### DISCUSSION TITLE
gnuradio osmocom source not present / gr-osmosdr not installable

### DISCUSSION THREAD

**Discussion by maichai** [Homebrew/discussions#1658] [Everyday usage]
I installed gnuradio and can also open the companion, however, I cannot really use it as the osmocom source is not there, which would let you receive RF using RTL-SDR, HackRF, etc.

So i wanted to install gr-osmosdr and I get: gr-osmosdr has been disabled because it does not build!

I can't haz nice things!

When will we get gr-osmosdr? I do haste librtlsdr and hackrf installed, so no problem on that front.

**Comment by danielnachun** [comment]
It appears that `gr-osmosdr` doesn't support gnuradio 3.9: https://github.com/Homebrew/homebrew-core/pull/69326.  The upstream source has an open ticket for this to be fixed: https://osmocom.org/issues/5021, but it currently remains unresolved.  I'm afraid there isn't much we can do until they've fixed this.

**Reply by maichai** [reply]
> It appears that `gr-osmosdr` doesn't support gnuradio 3.9: [Homebrew/homebrew-core#69326](https://github.com/Homebrew/homebrew-core/pull/69326). The upstream source has an open ticket for this to be fixed: https://osmocom.org/issues/5021, but it currently remains unresolved. I'm afraid there isn't much we can do until they've fixed this.

well an older version of gnuradio may be more helpful than a crippled new version? Could we go back to a version that works together with gr-osmosdr?

**Reply by SMillerDev** [reply]
> well an older version of gnuradio may be more helpful than a crippled new version? 

Not for the other software that depends on it.
> Could we go back to a version that works together with gr-osmosdr?

No, because that involves crippling a lot of other software. The better way is to help the gr-osmosdr developers add support for the new version. Because they will have to switch eventually either way.

**Reply by maichai** [reply]
I found someone who said he would fix it, you have a pointer/URL for the problem report that I may give him?

He pointed out that Master was ported to 3.9 gnuradio already in January and can be used. It seems they are not frequently pushing releases.

**Reply by cho-m** [reply]
The upstream https://osmocom.org/issues/5021 mentioned HEAD should be compatible with gnuradio@3.9:
> Since mid-january 2021 the master branch is for 3.9 and builds just fine - i'm not aware of any remaining issues?

Possible options to pursue:
1. continue proceeding along `brew install --head ...` route in other thread
2. check if `gnuradio@3.8` meets requirements as keg-only/versioned formula https://docs.brew.sh/Versions. If so, it can be added along with modifying dependency for `gr-osmocom`.

**Reply by danielnachun** [reply]
I guess it was never clarified in the issue I posted above if gnu radio is still actively maintaining 3.8.  It looks like the biggest problem here is just that `gr-osmosdr` upstream needs to tag a new release with the fixes.

**Comment by carlocab** [comment]
You can try `brew edit gr-osmosdr`, remove the `disable!` line, and then `brew install --HEAD gr-osmosdr`. Note: this requires building from source, which we don't support. We can't do much to help you if your build fails.

**Reply by maichai** [reply]
OK I tried that and installation runs through. However, when trying to use it, I get Library not loaded: /usr/local/opt/gnuradio/lib/libgnuradio-uhd.3.8.2.dylib

**Reply by carlocab** [reply]
Try `brew uninstall gr-osmosdr && brew install --HEAD gr-osmosdr`. If that still doesn't work then I'm unable to help you further; sorry.

**Reply by maichai** [reply]
RuntimeError: Will not overwrite /usr/local/etc/gnuradio/plugins.d/gr-osmosdr.pth
seems to be an old file, so I deleted it

Then the compilation and installation ran successfully.

It seems to work with gnuradio companion 3.9.1.0
However not successful with Universal Radio Hacker 2.9.2 when using the gnuradio backend. I wrote a PR for URH accordingly.

**Reply by danielnachun** [reply]
That appears to be more of a driver error?  Though it's hard to say given how many layers of software are at work here.
